### PR TITLE
Add flag for OpenSSL 1.1.0 compatibility

### PR DESCRIPTION
**Pull request by alanking**
On platforms which use OpenSSL 3.0, many functions and types have been
deprecated or hidden which are still in use. This adds a flag to make
these functions and types visible for building in such a way that is
compatible with 1.1.0.

See the OpenSSL documentation for more information:
https://www.openssl.org/docs/man3.0/man7/openssl_user_macros.html#OPENSSL_API_COMPAT

---

Still need to make sure this is okay on the older platforms which do not need this

### CODE REVIEW

**Comment by alanking** [comment]
The other approach is to, of course, replace the deprecated/missing functions and types, but this is the most expedient and possibly backward-compatible solution as I see it.

**Comment by trel** [comment]
i think this is the right answer for today.

**Comment by alanking** [comment]
We should probably try this on all of our supported platforms before merging. I think Centos 7 may still be using 1.0.x.

**Comment by JustinKyleJames** [comment]
I will test this out on the various platforms.

**Comment by alanking** [comment]
I have confirmed that libs3 builds on Centos 7 and Ubuntu 22 with this change. Haven't built S3 plugin or run tests, but I don't think we need to? Thoughts?

File: GNUmakefile [hunk 1]
```diff
@@ -150,6 +150,7 @@ CFLAGS += -Wall -Werror -Wshadow -Wextra -Iinc \
           -D__STRICT_ANSI__ \
           -D_ISOC99_SOURCE \
           -D_POSIX_C_SOURCE=200112L \
+		  -DOPENSSL_API_COMPAT=0x10100000L \
```
Comment (alanking) [inline GNUmakefile]: This worked fine for all the platforms that I built.

However, iRODS itself is not accepting this syntax for some platforms (specifically, almalinux:8, ubuntu:18.04, and debian:11). I had to use a different syntax to build iRODS packages for all platforms:
```make
-DOPENSSL_API_COMPAT=0x101000 \
```
This is acceptable according to the OpenSSL documentation. Should we use a similar syntax here, or just go with the current one which I have confirmed (i.e. the `0x10100000L` syntax)?

Comment (alanking) [reply GNUmakefile]: Actually, this "second form" is recommended in the docs. I'll change it here too.

Comment (alanking) [reply GNUmakefile]: Done

**Comment by alanking** [comment]
Force-push due to rebase.

And... it seems that this change does not play well with Ubuntu 22.04 (at least). For some reason, only the old form works here. Guess I should look into that...

**Comment by alanking** [comment]
After some discussion, we decided that using the "old" form is fine here. I removed the new commit so we're back where we started.

**Comment by alanking** [comment]
Created #27 for linking. Please review comment for comprehensibility. Awaiting @JustinKyleJames confirmation

**Comment by alanking** [comment]
Latest changes build for all platforms using the recommended form of the compatibility macro.

See https://github.com/irods/irods/issues/7175#issuecomment-1625690251 for more.

**Comment by trel** [comment]
so good.

**Comment by JustinKyleJames** [comment]
I pointed the S3 resource plugin to this version of libs3 and ran through the Test_S3_NoCache_V4 tests on Ubuntu 18, Ubuntu 20, and AlmaLinux 8.  All passed.

**Comment by alanking** [comment]
Any other thoughts? Will add # and merge on approval

**Pull request merged**